lib/components/term.js

```diff
@@ -143,39 +124,64 @@ export default class Term extends React.PureComponent {
     this.termWrapperRef.appendChild(this.termRef);
 
     if (!props.term) {
+      let needTransparency = Color(props.backgroundColor).alpha() < 1;
+      let useWebGL = false;
+      if (props.webGLRenderer) {
+        if (needTransparency) {
+          // eslint-disable-next-line no-console
+          console.warn(
+            'WebGL Renderer has been disabled since it does not support transparent backgrounds yet. ' +
+              'Falling back to canvas-based rendering.'
+          );
+        } else if (!isWebgl2Supported()) {
+          // eslint-disable-next-line no-console
+          console.warn('WebGL2 is not supported on your machine. Falling back to canvas-based rendering.');
+        } else {
+          // Experimental WebGL renderer needs some more glue-code to make it work on Hyper.
+          // useWebGL = true;
+        }
+      }
+      Term.reportRenderer(props.uid, useWebGL ? 'WebGL' : 'Canvas');
+
       this.term.attachCustomKeyEventHandler(this.keyboardHandler);
+      this.term.loadAddon(fitAddon);
+      this.term.loadAddon(searchAddon);
+      this.term.loadAddon(webLinksAddon);
       this.term.open(this.termRef);
-      this.term.webLinksInit();
-      this.term.winptyCompatInit();
+      fitAddon.fit();
+      if (useWebGL) {
+        this.term.loadAddon(webGlAddon);
+      }
+      // this.term.winptyCompatInit();
     }
 
     if (this.props.isTermActive) {
       this.term.focus();
     }
 
     if (props.onTitle) {
-      this.disposableListeners.push(this.term.addDisposableListener('title', props.onTitle));
+      this.disposableListeners.push(this.term.onTitleChange(props.onTitle));
     }
 
     if (props.onActive) {
-      this.disposableListeners.push(this.term.addDisposableListener('focus', props.onActive));
+      this.term.textarea.addEventListener('focus', props.onActive);
```

Comment:
You can still use `disposableListeners` and keep it all here if you want:

```ts
this.term.textarea.addEventListener('focus', props.onActive);
this.disposableListeners.push({ dispose: () =>this.term.textarea.removeEventListener('focus', this.props.onActive) });
```

Reply:
yep, did something similar on my fork, but this looks much cleaner